test/test-udp-alloc-recv-stop.c

```diff
@@ -0,0 +1,81 @@
+/* Copyright libuv project contributors. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+
+#include "task.h"
+#include "uv.h"
+
+static uv_udp_t server;
+static uv_udp_t client;
+
+
+static void fail_cb(uv_udp_t* handle,
+                    ssize_t nread,
+                    const uv_buf_t* buf,
+                    const struct sockaddr* addr,
+                    unsigned flags) {
+/* win will always call this cb so run the assertion in other platforms */
```

Comment:
Wait what? The asymmetry in this API is not something we want. Is thre no way around it? Is this perhaps the 0 length read windows does?

Reply:
I think it is because sanitizers (IIRC ASAN and TSAN) in win. Before that I just had a `void cb(void)` casted as `uv_udp_recv_cb`, zero problems in NIXes and some Win jobs, but the sanitizers one on win will call that cb.

Reply:
A, never mind, [you're right](https://github.com/libuv/libuv/actions/runs/13621777899).

Not quite sure if that asymmetry is in this PR scope. Seems to be related to something else.